### PR TITLE
Make empty objects and arrays collapsed in pretty print mode

### LINKED ISSUE #2502: Pretty Printing an empty class adds a newline to json object
**Describe the bug**

Empty classes get a newline character added to the json object:

```
{
  "some_empty_class": {
  }
}
```
It's not very pretty 😅

**To Reproduce**
```kotlin
@Serializable
data class Test(
    @SerialName("some_empty_class") val someEmptyClass: SomeEmptyClass,
)

@Serializable
class SomeEmptyClass

println(Json{prettyPrint = true}.encodeToString(Test(SomeEmptyClass())))
```

**Expected behavior**
All other libraries I have used when pretty printing would output:
```
{
  "some_empty_class": {}
}
```

**Environment**
 - Kotlin version: [e.g. 1.9.20]
 - Library version: [e.g. 1.6.0]

### PR DESCRIPTION
**Pull request by sandwwraith**
because it is a widespread convention.

Fixes #2502

### CODE REVIEW

File: formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonPrettyPrintTest.kt [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    class B(val prefix: String = "a", val empty: Empty = Empty(), val postfix: String = "b")
+
+    @Serializable
+    class Recursive(val rec: Recursive?, val empty: Empty = Empty())
+
+    @Serializable
+    class WithListRec(val rec: WithListRec?, val l: List<Int> = listOf())
+
+    @Test
+    fun testPlain() = parametrizedTest { mode ->
```
Comment (shanshin) [inline formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonPrettyPrintTest.kt:29]: Also test `Empty` on top-level?

File: formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonPrettyPrintTest.kt [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    class Empty
+
+    @Serializable
+    class A(val empty: Empty = Empty())
+
+    @Serializable
+    class B(val prefix: String = "a", val empty: Empty = Empty(), val postfix: String = "b")
+
+    @Serializable
+    class Recursive(val rec: Recursive?, val empty: Empty = Empty())
```
Comment (shanshin) [inline formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonPrettyPrintTest.kt:23]: Do we have tests on `encodeDefaults = false` when all properties are defaults?

**Review by shanshin — APPROVED** [review]

File: formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonPrettyPrintTest.kt (hunk 1 above)
Comment (sandwwraith) [reply formats/json-tests/commonTest/src/kotlinx/serialization/json/JsonPrettyPrintTest.kt:44]: Added tests
